pkgs/build-support/trivial-builders.nix

```
@@ -322,6 +322,67 @@ rec {
     $CC -x c code.c -o "$n"
     '';
 
+
+  /* concat a list of files to the nix store.
+   * The contents of files are added to the file in the store.
+   *
+   * Examples:
+   * # Writes my-file to /nix/store/<store path>
+   * concatTextFile {
+   *   name = "my-file";
+   *   files = [ drv1 "${drv2}/path/to/file" ];
+   * }
+   * # See also the `concatText` helper function below.
+   *
+   * # Writes executable my-file to /nix/store/<store path>/bin/my-file
+   * concatTextFile {
+   *   name = "my-file";
+   *   files = [ drv1 "${drv2}/path/to/file" ];
+   *   executable = true;
+   *   destination = "/bin/my-file";
+   * }
+   */
+  concatTextFile =
+    { name # the name of the derivation
+    , files
+    , executable ? false # run chmod +x ?
+    , destination ? ""   # relative path appended to $out eg "/bin/foo"
+    , checkPhase ? ""    # syntax checks, e.g. for scripts
+    , meta ? { }
+    }:
+    runCommandLocal name
+      { inherit files executable checkPhase meta destination; }
+      ''
+        file=$out$destination
```

```suggestion
        file=$out/${destination:-$name}
```

This is better because it makes this work:

```
runCommand "test" {} ''
cp ${concatText "foo.bar" []} .
mkdir -p $out
mv foo.bar $out/
''
```

Please add it as a test case; see `pkgs/tests/trivial-builders.nix`.

---

Sorry, this is only feasible after #119942, so you can set `outPath` correctly. Carry on, we'll improve it later. Please do add a simpler test though.